src/main/java/seedu/exercise/logic/commands/DeleteCommand.java

```diff
@@ -1,53 +1,25 @@
 package seedu.exercise.logic.commands;
 
-import static java.util.Objects.requireNonNull;
-
-import java.util.List;
-
-import seedu.exercise.commons.core.Messages;
-import seedu.exercise.commons.core.index.Index;
-import seedu.exercise.logic.commands.exceptions.CommandException;
-import seedu.exercise.model.Model;
-import seedu.exercise.model.exercise.Exercise;
+import static seedu.exercise.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static seedu.exercise.logic.parser.CliSyntax.PREFIX_INDEX;
+import static seedu.exercise.logic.parser.CliSyntax.PREFIX_NAME;
 
 /**
- * Deletes a exercise identified using it's displayed index from the exercise book.
+ * Represents a DeleteCommand with hidden internal logic and the ability to be executed.
  */
-public class DeleteCommand extends Command {
-
+public abstract class DeleteCommand extends Command {
     public static final String COMMAND_WORD = "delete";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the exercise identified by the index number used in the displayed exercise list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
-
-    public static final String MESSAGE_DELETE_EXERCISE_SUCCESS = "Deleted Exercise: %1$s";
+    public static final String MESSAGE_USAGE_EXERCISE = "Parameters: INDEX (must be a positive integer)\n"
+            + "\t\tExample: " + COMMAND_WORD + " " + PREFIX_CATEGORY + "exercise " + PREFIX_INDEX + "1";
 
-    private final Index targetIndex;
+    public static final String MESSAGE_USAGE_REGIME = "Parameters: REGIME NAME, INDEX (must be a positive integer)\n"
+            + "\t\tExample: " + COMMAND_WORD + " " + PREFIX_CATEGORY + "regime "
+            + PREFIX_NAME + "level 1 " + PREFIX_INDEX + "1";
```

Comment:
Same here. The `MESSAGE_USAGE` should belong to the specific classes

Reply:
ok